### PR TITLE
Clean up hanging references

### PR DESCRIPTION
**Pull request by KonajuGames**
This PR fixes up some hanging references held by static fields after Game.Exit was called.

Git normalized some inconsistent line endings on commit.

### CODE REVIEW

File: MonoGame.Framework/Windows/WinFormsGamePlatform.cs [hunk 1]
```diff
@@ -66,34 +66,34 @@ 1. Definitions
 */
 #endregion License
 
-using System;
-using System.Collections.Generic;
+using System;
+using System.Collections.Generic;
```
Comment (KonajuGames) [inline MonoGame.Framework/Windows/WinFormsGamePlatform.cs]: Git normalized line endings.

File: MonoGame.Framework/Windows/WinFormsGamePlatform.cs [hunk 2]
```diff
@@ -66,34 +66,34 @@ 1. Definitions
...
-using System.Windows.Forms;
-using Microsoft.Xna.Framework.Input;
-using XnaKeys = Microsoft.Xna.Framework.Input.Keys;
-
-
+using System.Windows.Forms;
+using Microsoft.Xna.Framework.Input;
+using XnaKeys = Microsoft.Xna.Framework.Input.Keys;
+
+
```
Comment (KonajuGames) [inline MonoGame.Framework/Windows/WinFormsGamePlatform.cs]: Git normalized line endings.

File: MonoGame.Framework/Windows/WinFormsGamePlatform.cs [hunk 3]
```diff
@@ -66,34 +66,34 @@ 1. Definitions
...
 namespace MonoGame.Framework
 {
     class WinFormsGamePlatform : GamePlatform
     {
         //internal static string LaunchParameters;
 
-        private WinFormsGameWindow _window;
+        private WinFormsGameWindow _window;
```
Comment (KonajuGames) [inline MonoGame.Framework/Windows/WinFormsGamePlatform.cs]: Git normalized line endings.

File: MonoGame.Framework/GamePlatform.cs [hunk 4]
```diff
@@ -337,6 +337,9 @@ protected virtual void Dispose(bool disposing)
         {
             if (!disposed)
             {
+                Mouse.PrimaryWindow = null;
+                TouchPanel.PrimaryWindow = null;
+
```
Comment (KonajuGames) [inline MonoGame.Framework/GamePlatform.cs]: Release the static reference from Mouse and TouchPanel.

File: MonoGame.Framework/Windows/WinFormsGamePlatform.cs [hunk 5]
```diff
@@ -247,10 +247,11 @@ protected override void Dispose(bool disposing)
...
                     _window.Dispose();
                     _window = null;
-                    Window = null;
-                }
-                Microsoft.Xna.Framework.Media.MediaManagerState.CheckShutdown();
+                    Window = null;
+                }
+                Microsoft.Xna.Framework.Media.MediaManagerState.CheckShutdown();
             }
+            Keyboard.SetKeys(null);
```
Comment (KonajuGames) [inline MonoGame.Framework/Windows/WinFormsGamePlatform.cs]: We set the List on Keyboard, so we should release it.

File: MonoGame.Framework/Windows/WinFormsGameWindow.cs [hunk 6]
```diff
@@ -67,7 +67,7 @@ class WinFormsGameWindow : GameWindow, IDisposable
         static private ReaderWriterLockSlim _allWindowsReaderWriterLockSlim = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
         static private List<WinFormsGameWindow> _allWindows = new List<WinFormsGameWindow>();
 
-        private readonly WinFormsGamePlatform _platform;
+        private WinFormsGamePlatform _platform;
```
Comment (KonajuGames) [inline MonoGame.Framework/Windows/WinFormsGameWindow.cs]: Cannot be readonly as we clear it in Dispose.

File: MonoGame.Framework/Windows/WinFormsGameWindow.cs [hunk 7]
```diff
@@ -508,6 +508,11 @@ void Dispose(bool disposing)
                     _form = null;
                 }
             }
+            _platform = null;
+            Game = null;
+            Mouse.SetWindows(null);
+            Device.KeyboardInput -= OnRawKeyEvent;
+            Device.RegisterDevice(UsagePage.Generic, UsageId.GenericKeyboard, DeviceFlags.Remove);
```
Comment (KonajuGames) [inline MonoGame.Framework/Windows/WinFormsGameWindow.cs]: Clear _platform and Game references.
Clear the windows list in Mouse.
Remove KeyboardInput event handler and try to unregister raw keyboard input.  This is not documented, but some forum posts suggest it is the right thing to do.

**Comment by KonajuGames** [comment]
I've commented the actual changes.

**Comment by tomspilman** [comment]
Seems reasonable... merging!

**Pull request merged**